src/main/java/net/dv8tion/jda/core/entities/Invite.java

```diff
@@ -367,4 +390,59 @@ default String getURL()
          */
         Set<String> getFeatures();
     }
+
+    /**
+     * POJO for the group information provided by an invite.
+     *
+     * @see #getChannel()
+     */
+    interface Group extends ISnowflake
+    {
+        /**
+         * The icon id of this group or {@code null} if the group has no icon.
+         *
+         * @return The group's icon id
+         *
+         * @see    #getIconUrl()
+         */
+        String getIconId();
+
+        /**
+         * The icon url of this group or {@code null} if the group has no icon.
+         *
+         * @return The group's icon url
+         *
+         * @see    #getIconId()
+         */
+        String getIconUrl();
+
+        /**
+         * The name of this group or {@code null} if the group has no name.
+         *
+         * @return The group's name
+         */
+        String getName();
+
+        /**
+         * The names of all users in this group. If the users were not included in the
+         * invite, this will return {@code null}. Users will usually only be returned when resolving the invite via the
+         * {@link #resolve(net.dv8tion.jda.core.JDA, java.lang.String, boolean) Invite.resolve()} method with the
+         * {@code withCounts} boolean set to {@code true}.
+         *
+         * @return The names of the groups's users or null if not preset in the invite
+         */
+        List<String> getUsers();
```

Comment:
Ok, I understand your intentions here for making it return null. What is your opinion on doing `Optional<List<String>>` instead?

Reply:
Do we have any other places in JDA where we return null or optional for a collection?
I feel like most of our non-collection returns that indicate missing data use `null`

Reply:
I can't think of any example

Reply:
Basically my point here is that we (for better or worse) don't use `Optional` in JDA when we lack data.

Are there any other places that we, upon not receiving a collection, just decide to do `List.emptyList`? For the most part I feel like we trying to not return null from collection related getters, so im conflicted here.

Reply:
I generally never return null when a collection is defined, so I can't think of any case where this would be done in JDA at the moment. Some indication of this being optional or possibly missing is definitely needed in my opinion, even if just an `@Nullable`.

Reply:
Personally I don't like switching to returning an empty object instead of `null` just because it's a collection. It's just feels unintuitive and inconsistent having if you have to make null-checks everywhere except for this specific case.